initial draft of lambda guidance (#2632)
open-telemetry/opentelemetry.io: content/en/docs/faas/_index.md

---

> title: Cloud Functions

Reviewer: For consistency call this "Functions as a Service", if this is to long for the sidebar menu you still can have a linkTitle like "Cloud Functions" or "FaaS"

description: >-

>   OpenTelemetry supports various methods of monitoring Cloud Functions

Reviewer: ```suggestion
  OpenTelemetry supports various methods of monitoring Functions as a Service provided by different cloud vendors
```

weight: 2
---

Functions as a Service (FaaS) is an important serverless compute platform for cloud native applications. However, platform quirks usually mean these applications have slightly different monitoring guidance and requirements than applications running on Kubernetes or Virtual Machines.

The initial vendor scope of the FaaS documentation is around Azure, GCP, and AWS functions (lambdas).

### Community Assets

> The community currently provides pre-built Lambda layers to quickly add instrumentation or the Collector to your Lambda.

Reviewer 2: This statement is a bit strange. 

> or the Collector to your Lambda.

I think you can say something like:

> The community currently provides pre-built Lambda layers able to auto-instrument your application as well as a the option of standalone collector lambda layer that can be used when instrumenting applications manually.
your Lambda.

Author: Tweaked the final wording slightly but updated

---
title: Functions as a Service
description: >-
  OpenTelemetry supports various methods of monitoring Functions as a Service
  provided by different cloud vendors

> weight: 2

Reviewer: Please update the weight in a way that FaaS docs comes after the instrumentation & collector

Author: How long does it take the preview to update?

---

Functions as a Service (FaaS) is an important serverless compute platform for
cloud native applications. However, platform quirks usually mean these
applications have slightly different monitoring guidance and requirements than
applications running on Kubernetes or Virtual Machines.

The initial vendor scope of the FaaS documentation is around Azure, GCP, and AWS

> functions (lambdas).

Reviewer: ```suggestion
functions (Lambdas).
```

you write Lambda with an uppercase L below, so I think this should be consistent

> The initial vendor scope of the FaaS documentation is around Azure, GCP, and AWS
> functions (Lambdas).

Reviewer: ```suggestion
The initial vendor scope of the FaaS documentation is around Microsoft Azure, Google Cloud Platform (GCP), and Amazon Web Services (AWS). AWS functions are also known as Lambda.
```

### Community Assets

> The community currently provides pre-built Lambda layers able to auto-instrument

Reviewer: ```suggestion
The OpenTelemetry community currently provides pre-built Lambda layers able to auto-instrument
```

your application as well as a the option of standalone Collector Lambda layer
that can be used when instrumenting applications manually or automatically.

The release status can be tracked in the
[OpenTelemetry-Lambda repo](https://github.com/open-telemetry/opentelemetry-lambda).

- [Get started with the Collector Lambda Layer](./lambda-collector)
- [Auto-instrumentation](./lambda-auto-instrument)

> - ]Manual instrumentation](./lambda-manual-instrument)

Reviewer: ```suggestion
- [Manual instrumentation](./lambda-manual-instrument)
```

---

> title: Functions as a Service

Reviewer 3: ```suggestion
title: Functions as a Service
linkTitle: FaaS
```

description: >-

>   OpenTelemetry supports various methods of monitoring Functions as a Service
>   provided by different cloud vendors

Reviewer 3: ```suggestion
  OpenTelemetry supports various methods of monitoring Function-as-a-Service
  provided by different cloud vendors
```

weight: 12
---

Functions as a Service (FaaS) is an important serverless compute platform for
cloud native applications. However, platform quirks usually mean these
applications have slightly different monitoring guidance and requirements than
applications running on Kubernetes or Virtual Machines.

The initial vendor scope of the FaaS documentation is around Microsoft Azure,
Google Cloud Platform (GCP), and Amazon Web Services (AWS). AWS functions are
also known as Lambda.

### Community Assets

The OpenTelemetry community currently provides pre-built Lambda layers able to
auto-instrument your application as well as a the option of standalone Collector
Lambda layer that can be used when instrumenting applications manually or
automatically.

The release status can be tracked in the
[OpenTelemetry-Lambda repository](https://github.com/open-telemetry/opentelemetry-lambda).

> 
> - [Get started with the Collector Lambda Layer](./lambda-collector)
> - [Auto-instrumentation](./lambda-auto-instrument)
> - [Manual instrumentation](./lambda-manual-instrument)

Reviewer 3: Drop these section page links since they are auto-generated by Hugo. 
```suggestion
```
